src/Culling/babylon.boundingBox.ts

```diff
@@ -117,14 +117,15 @@ module BABYLON {
          * @param factor defines the scale factor to apply
          * @returns the current bounding box
          */
+        @automaticPoolRelease(Tmp.PoolVector3)
         public scale(factor: number): BoundingBox {
-            const diff = Tmp.Vector3[0].copyFrom(this.maximum).subtractInPlace(this.minimum);
+            const diff = Tmp.PoolVector3.getTemp().copyFrom(this.maximum).subtractInPlace(this.minimum);

```

Comment:
Well the main problem is perf here. The math functions can be called thousands of times per frame. We do not want to add more code here.

I agree that TMP object can lead to errors and this is why we have MathTmp for instance for math functions (easier to track)

I would prefer having specific TMP objects per features (a static pool for bounding info would be fine) instead of adding more code processing to central functions

Reply:
yes, it would be simpler indeed to make sure each class deals with its own pool. And better for performance